idunn/utils/app.py

```diff
@@ -0,0 +1,59 @@
+from apistar import App
+
+class IdunnApp(App):
+    def __call__(self, environ, start_response):
+        def get_default_state():
+            return {
+                'environ': environ,
+                'start_response': start_response,
+                'exc': None,
+                'app': self,
+                'path_params': None,
+                'route': None,
+                'response': None,
+            }
+
+        state = get_default_state()
+        method = environ['REQUEST_METHOD'].upper()
+        path = environ['PATH_INFO']
+
+        if self.event_hooks is None:
+            on_request, on_response, on_error = [], [], []
+        else:
+            on_request, on_response, on_error = self.get_event_hooks()
+
+        try:
+            route, path_params = self.router.lookup(path, method)
+            state['route'] = route
+            state['path_params'] = path_params
+            if route.standalone:
+                funcs = [route.handler]
+            else:
+                funcs = (
+                    on_request +
+                    [route.handler, self.render_response] +
+                    on_response +
+                    [self.finalize_wsgi]
+                )
+            return self.injector.run(funcs, state)
+        except Exception as exc:
+            # This is the only change with the parent class.
```

Comment:
Instead of keeping everything inside this exception block, since we return from there, why not remove one indent level and put `pass` and some logging into the exception block instead?

Reply:
I prefer to stick to the parent class implementation if you don't mind.
And `exc` is used in the block, so it looks more explicit this way.

Reply:
I really don't like the fact that you have inner exception blocks in it. What about a function call instead?

Reply:
it should look cleaner now